app/services/katello/repository_type_manager.rb

```diff
@@ -140,6 +140,19 @@ def enabled?(repository_type)
         find(repository_type).present?
       end
 
+      def generic_remote_options(opts = {})
+        options = []
+        repo_types = opts[:enabled_only] ? enabled_repository_types : defined_repository_types
```

Comment:
You might want to make the default use `enabled_repository_types` if no options are given. I think in most cases you will want to used the enabled types only.